src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -274,69 +274,87 @@ private void LoadThreads()
             }
         }
 
-        private int GetReadableBytesCount(ulong address, int bytesRequested)
+        private int GetReadableBytesCount(ulong address, int bytesToRead)
         {
-            if (bytesRequested < 1)
+            if (bytesToRead <= 0)
             {
                 return 0;
             }
 
-            ulong endAddress = address + (ulong)bytesRequested - 1;
-            int bytesReadable = 0;
-            ulong prevEndAddr = default;
+            int i = GetMapEntryContaining(address);
+            if (i < 0)
+            {
+                return 0;
+            }
 
-            int startIndex = -1;
-            for (int i = 0; i < _memoryMapEntries.Count; i++)
+            int bytesReadable;
+            ulong prevEndAddr;
             {
                 MemoryMapEntry entry = _memoryMapEntries[i];
-                ulong entryBeginAddr = entry.BeginAddress;
                 ulong entryEndAddr = entry.EndAddress;
-                if (entryBeginAddr <= address && address < entryEndAddr && entry.IsReadable)
-                {
-                    int regionSize = (int)(entryEndAddr - address);
-                    if (regionSize >= bytesRequested)
-                    {
-                        return bytesRequested;
-                    }
 
-                    startIndex = i;
-                    bytesRequested -= regionSize;
-                    bytesReadable = regionSize;
-                    prevEndAddr = entryEndAddr;
-                    break;
+                int regionSize = (int)(entryEndAddr - address);
+                if (regionSize >= bytesToRead)
+                {
+                    return bytesToRead;
                 }
-            }
 
-            if (startIndex < 0)
-            {
-                return 0;
+                bytesToRead -= regionSize;
+                bytesReadable = regionSize;
+                prevEndAddr = entryEndAddr;
             }
 
-            for (int i = startIndex + 1; i < _memoryMapEntries.Count; i++)
+            for (i += 1; i < _memoryMapEntries.Count; i++)
```

Comment:
This block of code looks similar (or a duplicate of) the code in `MacOSProcessDataReader.cs` (I didn't look very closely to confirm they are the same).

If these are the same, is there any way we could put this in a static helper method, extension method, etc, so that we don't have it duplicated?  I don't like having two copies of the same algorithm that have to be updated in parallel.